### PR TITLE
Base druid-tree layout on 0010 OCFL extension

### LINKED ISSUE #13: Create and use a druid-pairtree extension
Blocked by #12 

Once [storage root support](#12) is added, consider creating and adding support for using [community extensions](https://ocfl.github.io/extensions/) such as the druid-based pairtree directory layout we currently use in stacks, and want to keep supporting in the OCFL structure. The extension should be used by the gem to navigate from the storage root to an object root using our druid-based pairtree hierarchy. For inspiration, review other production OCFL libraries, e.g.:

* https://github.com/OCFL/ocfl-java/blob/main/docs/USAGE.md#ocfl-repository-configuration
* https://github.com/inveniosoftware/ocflcore#quick-start

### PR DESCRIPTION
**Pull request by mjgiarlo**
Blocked by OCFL/extensions#76
Fixes #13

This commit extends `StorageRoot#save` to write a copy of the OCFL spec (as markdown, included in this commit) to the storage root---so it is self-describing as recommended in the spec---and changes the DruidTree layout so that it writes `ocfl_layout.json` in the storage root to make it clear which object root layout is used and also specifies that the DruidTree layout is based on a community extension (0010, currently in progress).

It also includes binstubs as an unrelated development touch-up (in its own commit).

### CODE REVIEW

File: lib/ocfl/layouts/druid_tree.rb [hunk 1]
```diff
@@ -16,6 +31,37 @@ def self.path_to(identifier)
...
+      end
+
+      def extension_directory
+        base_directory / EXTENSIONS_DIRECTORY / extension_name
+      end
+
+      def to_config_json
+        JSON.generate(
+          {
+            extensionName: "0010-differential-n-tuple-omit-prefix-storage-layout",
```
Comment (aaron-collier) [inline lib/ocfl/layouts/druid_tree.rb:48]: Should this use the `extension_name` method as it is on line 59?

Comment (mjgiarlo) [reply lib/ocfl/layouts/druid_tree.rb:48]: 💯, thanks!

File: .rubocop.yml [hunk 2]
```diff
@@ -5,6 +5,9 @@ require:
 
 AllCops:
   TargetRubyVersion: 3.1
+  Exclude:
+    - bin/**/*
+    - tmp/**/*
```
Comment (mjgiarlo) [inline .rubocop.yml:10]: Don't bother scanning generated binstubs or anything in `tmp/` (such as fake storage roots used in dev/test which may contain Ruby files).

File: README.md [hunk 3]
```diff
@@ -93,9 +93,9 @@ object.path(filepath: "ocfl.rbs")
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+After checking out the repo, run `bin/setup` to install dependencies. Then, run `bin/rake` to run the tests and linters. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
 
-To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and the created tag, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+To install this gem onto your local machine, run `bin/rake install`. To release a new version, update the version number in `version.rb`, and then run `bin/rake release`, which will create a git tag for the version, push git commits and the created tag, and push the `.gem` file to [rubygems.org](https://rubygems.org).
```
Comment (mjgiarlo) [inline README.md:98]: Prefer using binstubs to manual `bundle exec` which is needed only in some Ruby version management environments.

File: lib/ocfl.rb [hunk 4]
```diff
@@ -17,6 +17,11 @@
 module OCFL
   class Error < StandardError; end
 
+  module Extensions
+    CONFIG_FILE = "config.json"
+    STORAGE_ROOT_SUBDIRECTORY = "extensions"
+  end
```
Comment (mjgiarlo) [inline lib/ocfl.rb:23]: Values are shared by all OCFL extensions.

File: .rubocop.yml [hunk 5]
```diff
@@ -385,3 +388,7 @@ RSpec/UndescriptiveLiteralsDescription: # new in 2.29
   Enabled: true
 RSpec/VerifiedDoubleReference: # new in 2.10.0
   Enabled: true
+Style/SendWithLiteralMethodName: # new in 1.64
+  Enabled: true
+Style/SuperArguments: # new in 1.64
+  Enabled: true
```
Comment (mjgiarlo) [inline .rubocop.yml:394]: Add missing cops.

File: spec/support/temp_directory.rb [hunk 6]
```diff
@@ -6,7 +6,7 @@
   # Create an OCFL storage root
   around do |example|
     Dir.mktmpdir("ocfl-rspec-") do |base_directory|
-      @base_directory = base_directory
+      @base_directory = Pathname.new(base_directory)
```
Comment (mjgiarlo) [inline spec/support/temp_directory.rb:9]: Makes for smoother path construction plus the gem also uses pathnames

File: .rubocop.yml (hunk 2 above)
Comment (mjgiarlo) [inline .rubocop.yml:10]: This broke the build. I'm going to tweak this in the morning.

**Comment by mjgiarlo** [comment]
@aaron-collier Mind having another look-see?

**Review by jcoyne — APPROVED** [review]

**Pull request merged**